schema/deploy/tables/ciip_user.sql

```diff
@@ -3,83 +3,57 @@
 
 begin;
 
-create table ggircs_portal.ciip_user
-(
-  id integer primary key generated always as identity,
-  uuid uuid not null,
-  first_name varchar(1000),
-  last_name varchar(1000),
-  email_address varchar(1000),
-  occupation varchar(1000),
-  phone_number varchar(1000),
-  created_at timestamp with time zone not null default now(),
-  created_by int references ggircs_portal.ciip_user,
-  updated_at timestamp with time zone not null default now(),
-  updated_by int references ggircs_portal.ciip_user,
-  deleted_at timestamp with time zone,
-  deleted_by int references ggircs_portal.ciip_user
-
-);
-
-create trigger _100_timestamps
-  before insert or update on ggircs_portal.ciip_user
-  for each row
-  execute procedure ggircs_portal_private.update_timestamps();
-
-create unique index user_email_address_uindex
-  on ggircs_portal.ciip_user(email_address);
-
-  create unique index user_email_address_uuuid
-  on ggircs_portal.ciip_user(uuid);
-
-create trigger _welcome_email
-  after insert on ggircs_portal.ciip_user
-  for each row
-  execute procedure ggircs_portal_private.run_graphile_worker_job('welcome');
+-- Disabling trigger before updating data in-place
+alter table ggircs_portal.ciip_user disable trigger _100_timestamps;
+
+-- Changing uuid to type varchar
+
+-- ~~~NOTE FOR DEVELOPERS~~~
+-- We are not renaming this field to `session_sub` or something more appropriate, just due to the
+-- the amount of refactoring work this would require.
+-- As of this migration, it's implicit that the 'uuid' means 'the authentication provider's unique identifier'
+alter table ggircs_portal.ciip_user alter column uuid type varchar(1000);
+
+alter index ggircs_portal.user_email_address_uuuid rename to user_provider_uuid;
+
+-- Adding allow_uuid_update column
+alter table ggircs_portal.ciip_user
+  add column allow_uuid_update boolean not null default false;
+
+create trigger user_uuid_immutable_with_flag
+    before update of uuid on ggircs_portal.ciip_user
+    for each row
+    execute function ggircs_portal_private.user_uuid_immutable_with_flag_set();
 
 do
 $grant$
 begin
--- Grant ciip_administrator permissions
-perform ggircs_portal_private.grant_permissions('select', 'ciip_user', 'ciip_administrator');
-perform ggircs_portal_private.grant_permissions('insert', 'ciip_user', 'ciip_administrator');
-perform ggircs_portal_private.grant_permissions('update', 'ciip_user', 'ciip_administrator',
-  ARRAY['first_name', 'last_name', 'email_address', 'occupation', 'phone_number', 'created_at', 'created_by', 'updated_at', 'updated_by', 'deleted_at', 'deleted_by']);
 
--- Grant ciip_analyst permissions
-perform ggircs_portal_private.grant_permissions('select', 'ciip_user', 'ciip_analyst');
-perform ggircs_portal_private.grant_permissions('insert', 'ciip_user', 'ciip_analyst');
+-- Permissions for allow_uuid_update column
+perform ggircs_portal_private.grant_permissions('update', 'ciip_user', 'ciip_administrator',
+  ARRAY['uuid','allow_uuid_update']);
```

Comment:
maybe we don't need the update uuid permission since the function in charge of doing the update has a `security definer`

Reply:
confirming this